source/server/server.cc

```diff
@@ -710,6 +717,11 @@ void InstanceImpl::run() {
   // startup (see RunHelperTest in server_test.cc).
   const auto run_helper = RunHelper(*this, options_, *dispatcher_, clusterManager(),
                                     access_log_manager_, init_manager_, overloadManager(), [this] {
+                                      if (!startup_) {
+                                        notifyCallbacksForStage(Stage::Startup);
+                                        startup_ = true;
+                                      }
```

Comment:
Can you explain these changes? It's not clear to my why they were made. Please add more comments both here and below if they are necessary.

Reply:
This ensures Startup notifications to be sent first. Otherwise at  [LifecycleNotifications test](https://github.com/envoyproxy/envoy/pull/14319/files/871c415268f7a712fbb91fd9d7bccacafa3e0557#diff-d367e857aadddf9af71c8dfd9abfb3fc02ba147db56e1fc44ad41f60d1e9dd74R360) , the notification order will be `PostInit`, `WorkerStarted`, `Startup`, and deadlock will happen.(because we [block](https://github.com/envoyproxy/envoy/pull/14319/files/871c415268f7a712fbb91fd9d7bccacafa3e0557#diff-d367e857aadddf9af71c8dfd9abfb3fc02ba147db56e1fc44ad41f60d1e9dd74R382) callback at WorkerStarted stage)